src/spatiotemporalembedding.jl

```diff
@@ -208,31 +188,96 @@ function Base.show(io::IO, em::SpatioTemporalEmbedding{Φ,BC, X}) where {Φ,BC,X
     end
 end
 
+#####################################################################################
+#                                   CONSTRUCTORS                                    #
+#####################################################################################
+
+function SpatioTemporalEmbedding(
+		s::AbstractArray{<:AbstractArray{T,Φ}},
+		D, τ, B, k, boundary::BC
+		) where {T,Φ, BC<:AbstractBoundaryCondition}
+    if (BC <: ConstantBoundary) && typeof(boundary.c) != T
+	     throw(ArgumentError(
+		"Boundary value must be same element type as the timeseries data."))
+	end
+	X = (D+1)*(2B+1)^Φ
+	τs = Vector{Int}(undef,X)
+	βs = Vector{CartesianIndex{Φ}}(undef,X)
+	n = 1
+	for d=0:D, α = Iterators.product([-B*k:k:B*k for φ=1:Φ]...)
+		τs[n] = d*τ
+		βs[n] = CartesianIndex(α)
+		n +=1
+	end
+	return SpatioTemporalEmbedding{X}(τs, βs, boundary, size(s[1]))
+end
+
 
-"""
-	reconstruct(s::AbstractArray{<:AbstractArray{T,Φ}}, em)
-Reconstruct the spatial timeseries `s` represented by a `Vector` of `AbstractArray`
-states using the embedding struct `em` of type [`AbstractSpatialEmbedding`](@ref).
 
-Returns the reconstruction in the form of a [`Dataset`](@ref) where each row is a
-reconstructed state and they are ordered first through linear indexing into each state
-and then incrementing in time.
+function indices_within(radius, dimension)
+    #Return indices β within hypersphere with radius
+    #Floor radius to nearest integer. Does not lose points
+    r = floor(Int,radius)
+    #Hypercube of indices
+    hypercube = CartesianIndices((repeat([-r:r], dimension)...,))
+    #Select subset of hc which is in Hypersphere
+
+    βs = [ β for β ∈ hypercube if norm(β.I) <= radius ]
+end
+
 """
-function reconstruct(s::AbstractArray{<:AbstractArray{T,Φ}},
-	em::AbstractSpatialEmbedding{Φ,BC,X}
-	) where {T<:Number,Φ,BC,X}
-	timesteps = (length(s) - get_τmax(em))
-	num_pt    = get_num_pt(em)
-	L         = timesteps*num_pt
-
-	pt_in_space = CartesianIndices(s[1])
-	lin_idxs    = LinearIndices(s[1])
-	data = Vector{SVector{X,T}}(undef,L)
-	recv = zeros(T,X)
-	@inbounds for t in 1:timesteps, α in pt_in_space
-		n = (t-1)*num_pt+lin_idxs[α]
-		em(recv,s,t,α)
-		data[n] = recv
-	end
-	return Dataset(data)
+    LightConeEmbedding(s, N, τ, r₀, c, bc) → SpatioTemporalEmbedding
```

Comment:
please rename the camel case to either `light_cone_embedding` or much simpler `lightcone`. A camel-cased function indicates that it returns an instance of a type homonymous to the function. There is no `LightConeEmbedding` type.